cmd/kubebuilder-gen/internal/controllergen/inject.go

```diff
@@ -43,17 +43,21 @@ func (d *injectGenerator) Imports(c *generator.Context) []string {
 
 	repo := d.Controllers[0].Repo
 	im := []string{
-		"time",
 		"github.com/kubernetes-sigs/kubebuilder/pkg/controller",
 		"k8s.io/client-go/rest",
 		repo + "/pkg/controller/sharedinformers",
 		repo + "/pkg/client/informers/externalversions",
 		repo + "/pkg/inject/args",
 		"rbacv1 \"k8s.io/api/rbac/v1\"",
-		"k8s.io/client-go/kubernetes/scheme",
-		"rscheme " + "\"" + repo + "/pkg/client/clientset/versioned/scheme\"",
 	}
 
+	if len(d.APIS.Groups) > 0 {
+	    im = append(im, []string{
+	    	"time",
+	    	"k8s.io/client-go/kubernetes/scheme",
+	    	"rscheme " + "\"" + repo + "/pkg/client/clientset/versioned/scheme\""}...
+	    )
+    }
```

Comment:
Keeping track of these imports feels like a lot of hand-written magic :)
Another possible approach could be ...we add all the imports and then run `goimports` to remove extra imports.

Reply:
As discussed offline, Let `goimports` add imports could be risky. We may only allow `goimports` to remove extra imports. It can be done in a different PR.